### PR TITLE
MsgPack compliance for ledger Variant

### PR DESCRIPTION
**Pull request by troels-im**
This PR introduces MsgPack compliance for the ledger variant. As a consequence it is now possible to load either JSON or YAML and convert them into native C++ types through serialization, or use them them as arguments for an Etch script.

### CODE REVIEW

File: libs/dmlf/include/dmlf/execution/execution_result.hpp [hunk 1]
```diff
@@ -103,8 +103,17 @@ struct MapSerializer<fetch::dmlf::ExecutionResult, D>
   template <typename Constructor>
   static void Serialize(Constructor &map_constructor, Type const &exec_res)
   {
-    auto map = map_constructor(3);
-    map.Append(OUTPUT, exec_res.output_);
+    auto             map    = map_constructor(3);
+    variant::Variant output = exec_res.output_;
+
+    if (output.IsUndefined())
```
Comment (lrahmani) [inline libs/dmlf/include/dmlf/execution/execution_result.hpp:109]: the `IsUndefined` check should't be done at `ExecutionResult` serialization, but rather in the `variant::Variant` serializer/deserializer

File: libs/dmlf/tests/execution/local_executor_tests.cpp [hunk 2]
```diff
@@ -131,6 +131,8 @@ TEST_F(LocalExecutorTests, delete_state)
 
 TEST_F(LocalExecutorTests, create_executable)
 {
+  // TODO(tfr): It is unclear why running the same code twice would yeild two different results
```
Comment (lrahmani) [inline libs/dmlf/tests/execution/local_executor_tests.cpp]: executable names are unique (similarly to state names)

File: libs/dmlf/tests/execution/local_executor_tests.cpp [hunk 3]
```diff
@@ -131,6 +131,8 @@ TEST_F(LocalExecutorTests, delete_state)
...
+  // Some explanation would be good.
   result_promise = CreateExecutable("HelloWorld", SourceFile{"hello_world.etch", hello_world_etch});
```
Comment (lrahmani) [inline libs/dmlf/tests/execution/local_executor_tests.cpp:136]: executable names are unique (similarly to state names)

File: libs/dmlf/include/dmlf/execution/execution_result.hpp (hunk 1 above)
Comment (troels-im) [reply libs/dmlf/include/dmlf/execution/execution_result.hpp:109]: No, it should be done here - it does not belong in the serialisation.

File: libs/dmlf/tests/execution/local_executor_tests.cpp (hunk 3 above)
Comment (troels-im) [reply libs/dmlf/tests/execution/local_executor_tests.cpp:136]: Please add comments in the code to explain this as it is non-obvious when reading the test.

**Review by robertdickson — APPROVED** [review]
looks good to me

File: libs/dmlf/include/dmlf/execution/execution_result.hpp (hunk 1 above)
Comment (lrahmani) [reply libs/dmlf/include/dmlf/execution/execution_result.hpp:109]: in my understanding Undefined is a reflection of the variant being empty which is part of the variant state not the ExecutionResult

**Pull request merged**